### PR TITLE
Memory stats for cgroup-v2

### LINKED ISSUE #10190: Fedora 33 exec and java memory usage of allocation is always 0 bytes
### Nomad version
Version `1.0.4`

### Operating system and Environment details
Fedora 33
Kernel version 5.10.20-200.fc33.x86_64

```
awk '{print $1 " " $4}' /proc/cgroups
#subsys_name enabled
cpuset 1
cpu 1
cpuacct 1
blkio 1
memory 1
devices 1
freezer 1
net_cls 1
perf_event 1
net_prio 1
hugetlb 1
pids 1
```

### Issue

On Fedora 33 the java and exec drivers using the cgroup and chroot isolations shows 0 memory usage.

The stats object for the allocation looks as follows:

```json
{
    "ResourceUsage": {
        "MemoryStats": {
            "RSS": 0,
            "Cache": 0,
            "Swap": 1430994944,
            "Usage": 314494976,
            "MaxUsage": 0,
            "KernelUsage": 0,
            "KernelMaxUsage": 0,
            "Measured": [
                "RSS",
                "Cache",
                "Swap",
                "Usage",
                "Max Usage",
                "Kernel Usage",
                "Kernel Max Usage"
            ]
        },
        "CpuStats": {
            "SystemMode": 32.01995405139944,
            "UserMode": 14.847059837776385,
            "TotalTicks": 1218.5424242818265,
            "ThrottledPeriods": 0,
            "ThrottledTime": 0,
            "Percent": 46.86701631853179,
            "Measured": [
                "System Mode",
                "User Mode",
                "Throttled Periods",
                "Throttled Time",
                "Percent"
            ]
        },
        "DeviceStats": []
    },
    "Tasks": {
        "server": {
            "ResourceUsage": {
                "MemoryStats": {
                    "RSS": 0,
                    "Cache": 0,
                    "Swap": 1430994944,
                    "Usage": 314494976,
                    "MaxUsage": 0,
                    "KernelUsage": 0,
                    "KernelMaxUsage": 0,
                    "Measured": [
                        "RSS",
                        "Cache",
                        "Swap",
                        "Usage",
                        "Max Usage",
                        "Kernel Usage",
                        "Kernel Max Usage"
                    ]
                },
                "CpuStats": {
                    "SystemMode": 32.01995405139944,
                    "UserMode": 14.847059837776385,
                    "TotalTicks": 1218.5424242818265,
                    "ThrottledPeriods": 0,
                    "ThrottledTime": 0,
                    "Percent": 46.86701631853179,
                    "Measured": [
                        "System Mode",
                        "User Mode",
                        "Throttled Periods",
                        "Throttled Time",
                        "Percent"
                    ]
                },
                "DeviceStats": null
            },
            "Timestamp": 1615904472393702100,
            "Pids": {
                "105077": {
                    "MemoryStats": {
                        "RSS": 321540096,
                        "Cache": 0,
                        "Swap": 0,
                        "Usage": 0,
                        "MaxUsage": 0,
                        "KernelUsage": 0,
                        "KernelMaxUsage": 0,
                        "Measured": [
                            "RSS",
                            "Swap"
                        ]
                    },
                    "CpuStats": {
                        "SystemMode": 32.001905649477614,
                        "UserMode": 15.000890137819889,
                        "TotalTicks": 0,
                        "ThrottledPeriods": 0,
                        "ThrottledTime": 0,
                        "Percent": 47.00277673603846,
                        "Measured": [
                            "System Mode",
                            "User Mode",
                            "Percent"
                        ]
                    },
                    "DeviceStats": null
                }
            }
        }
    },
    "Timestamp": 1615904472393702100
}
```

CPU stats seem to be unaffected and running the same job on a ubuntu 20.04 installation shows the memory usage correctly.

### Reproduction steps

This can be reproduced with any simple example using the exec or java driver on a fedora 33 vagrant box.

#### Expected Result

Expecting the memory usage to be populated correctly. I found this previous issue [9120](https://github.com/hashicorp/nomad/issues/9120), but I don't think it's related.

### PR DESCRIPTION
**Pull request by notnoop**
When running with cgroup-v2, the measured memory stats are Usage, Cache, and Swap only: https://github.com/opencontainers/runc/blob/v1.0.0-rc93/libcontainer/cgroups/fs2/memory.go#L75-L106 .

Also, in RSS isn't measured in a task, then `nomad alloc status` will use the reported Usage for the top level item.  This PR only changes the CLI, and we need to update the UI as well.

Fixes #10190
Related to #10251

### CODE REVIEW

File: command/helpers.go [hunk 1]
```diff
@@ -542,3 +542,14 @@ func (w *uiErrorWriter) Close() error {
 	}
 	return nil
 }
+
+// stringsContains returns true if s is present in the vs string slice
+func stringsContain(vs []string, s string) bool {
```
Comment (tgross) [inline command/helpers.go:547]: There's a [`helper.SliceStringContains`](https://github.com/hashicorp/nomad/blob/v1.0.4/helper/funcs.go#L190-L198) that implements exactly this function. (It's perhaps inevitable in a code base this size that we have multiple implementations of it floating around 😁 )

File: drivers/docker/util/stats_posix.go [hunk 2]
```diff
@@ -12,17 +12,28 @@ import (
...
+
+	// cgroup-v2 only exposes a subset of memory stats
+	DockerCgroupV1MeasuredMemStats = []string{"RSS", "Cache", "Swap", "Usage", "Max Usage"}
+	DockerCgroupV2MeasuredMemStats = []string{"Cache", "Swap", "Usage"}
 )
 
 func DockerStatsToTaskResourceUsage(s *docker.Stats) *cstructs.TaskResourceUsage {
+	measuredMems := DockerCgroupV1MeasuredMemStats
+
+	// use a simple heauristic to check if cgroup-v2 is used.
```
Comment (tgross) [inline drivers/docker/util/stats_posix.go:24]:
```suggestion
	// use a simple heuristic to check if cgroup-v2 is used.
```

File: drivers/docker/util/stats_posix.go [hunk 3]
```diff
@@ -12,17 +12,28 @@ import (
 
 var (
 	DockerMeasuredCPUStats = []string{"Throttled Periods", "Throttled Time", "Percent"}
-	DockerMeasuredMemStats = []string{"RSS", "Cache", "Swap", "Usage", "Max Usage"}
+
+	// cgroup-v2 only exposes a subset of memory stats
+	DockerCgroupV1MeasuredMemStats = []string{"RSS", "Cache", "Swap", "Usage", "Max Usage"}
+	DockerCgroupV2MeasuredMemStats = []string{"Cache", "Swap", "Usage"}
```
Comment (tgross) [inline drivers/docker/util/stats_posix.go:18]: In https://github.com/hashicorp/nomad/issues/9073 we have an issue where metrics that aren't being measured are still being emitted to the metrics endpoint. Does this help with that situation here or will this still result in empty metrics being emitted? See [`task_runner.go#L1309-L1336`](https://github.com/hashicorp/nomad/blob/v1.0.4/client/allocrunner/taskrunner/task_runner.go#L1309-L1336).

Comment (notnoop) [reply drivers/docker/util/stats_posix.go:18]: This wouldn't address that issue.  We can update task_runners.go to see if the metrics are measured.

**Review by tgross — APPROVED** [review]
LGTM

**Pull request merged**